pyriemann/utils/kernel.py

```diff
@@ -0,0 +1,107 @@
+import numpy as np
+
+from .mean import mean_riemann
+from .base import invsqrtm, logm
+
+
+def kernel(X, Y=None, Cref=None, metric='riemann', reg=1e-10):
+    r""" Calculates the Kernel matrix K of inner products of two sets
+         X and Y of SPD matrices on tangent space of C according to the
+         specified metric.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_matrices, n_channels, n_channels)
+            First set of SPD matrices.
+        Y : None | ndarray, shape (n_matrices, n_channels, n_channels)
+            Second set of SPD matrices. If None, Y is set to X.
+        Cref : None | ndarray, shape (n_channels, n_channels)
+            Reference point for the tangent space and inner product
+            calculation. If None, Cref is calculated as the mean of X
+            according to the specified metric.
+        metric : {'riemann'}
```

Comment:
Can do, although I think the respective kernels look different when we start from the geodesics and do the derivation of tangent space projection using the logarithmic map and scalar product on tangent space <A,B>_P = tr(P^{-1/2}AP^{-1/2}P^{-1/2}BP^{-1/2}) there. 
If my calculations are correct, these are the resulting kernels:

For euclid, it would be 
K^P_{i,j} = {tr}(P^{-1/2}(X_i - P)P^{-1/2} P^{-1/2}(Y_i - P)P^{-1/2})

For logeuclid it would be 
K^P_{i,j} = {tr}((\log(A) - \log(P)(\log(B) - \log(P)).

It is quite possible that this gives kernel matrices with the same properties for RSVC and RLLE as compared to your kernels, but I'll have to look at that in a bit more detail.

Reply:
It gives different kernel matrices leading to different embeddings. The embeddings look a bit better with my kernels so I implemented it like this.

Reply:
Have a look to equations 7 and 10 of 
https://hal.archives-ouvertes.fr/file/index/docid/820475/filename/BARACHANT_Neurocomputing_ForHal.pdf

This article should be added as reference for Riemannian and LogEuclidean kernels.

Reply:
For the Riemannian kernel I can do that. 
My log-euclidean kernel is different, though. And I think more appropriate, as it takes as a basis the log-euclidean geodesic and goes from there. The log-euclidean kernel in the paper ist just a specific version of the Riemann kernel with C_ref=I. I can add that as another additional kernel, but would like to stick with my implementation of the log-euclidean kernel to keep it persistent with the geodesic.

Reply:
In Arsigny's papers, Log-Euclidean metric is defined as the (affine-invariant) Riemannian metric with Cref=Id.

I agree that it is always possible to add parameters to the equation, but it is not the canonical Log-Euclidean kernel anymore.

Reply:
I beg to differ. I'd say that my kernel is indeed the true canonical Log-Euclidean kernel as it is derived from the log-euclidean geodesic. 
The Log-Euclidean kernel in the paper is just a specific version of the Riemannian kernel which derived from the Riemannian geodesic. It is just a very unfortunate choice of naming.
I can provide the derivation if needed, I'll just have to write it up so it's readable (can do next week).

Reply:
Looking forward to your derivation!
Once again: ok for me to use a custom kernel in an example.
But, functions in pyRiemann modules must be canonical kernels, supported by published references.